numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1908,6 +1908,34 @@ error:
 }
 /**end repeat**/
 
+/**begin repeat
+ *  #name = half, float, double, longdouble#
+ *  #Name = Half, Float, Double, LongDouble#
+ *  #is_half = 1,0,0,0#
+ *  #c    = f, f, , l#
+ */
+static PyObject *
+@name@_is_integer(PyObject *self)
+{
+#if @is_half@
+    npy_double val = npy_half_to_double(PyArrayScalar_VAL(self, @Name@));
+#else
+    npy_@name@ val = PyArrayScalar_VAL(self, @Name@);
+#endif
+    PyObject *o;
+
+    if (npy_isnan(val)) {
+        Py_RETURN_FALSE;
+    }
+    if (!npy_isfinite(val)) {
+        Py_RETURN_FALSE;
+    }
+
+    o = (npy_floor@c@(val) == val) ? Py_True : Py_False;
+    Py_INCREF(o);
+    return o;
+}
```

Comment:
The implementation is largly based on its counterpart in cpython: https://github.com/python/cpython/blob/60b93d9e4922eeae25052bc15909d1f4152babde/Objects/floatobject.c#L853-L873

Reply:
Maybe `res` instead of `o`?

Reply:
Sure, why not. Done as of https://github.com/numpy/numpy/pull/19803/commits/bb3e4077a32078deb31d7c31034303303fc268f4.